gh-120397: improve the speed of str.count, bytes.count et al. for single characters by about 2x. (#120398)
python/cpython: Misc/NEWS.d/next/Core and Builtins/2024-06-12-13-47-25.gh-issue-120397.n-I_cc.rst

> Improve the :meth:`str.count`, :meth:`bytes.count` and :meth:`bytearray.count`

Reviewer: "Improve"? How? Please elaborate. I suppose that you mean "Improve the speed". It's also good to give an idea of the potential speed, for example: "up to 2x faster".

Reviewer 2: Ah that's my bad for this one. I suggested this wording but did not consider the "improve the speed" since I assumed that it would be put in the "optimizations" section of the whats new but it's not the case.

Reviewer 3: How much? Does it make it 10% faster, 10 times faster?

Author: Up to 2x I would say. At least on my x86-64 machine.

> Significantly improve the speed of the str.count, bytes.count and
> bytearray.count method when the argument is a single character and the
> target architecture and compiler support vectorization.

Reviewer 2: ```suggestion
Improve the :meth:`str.count`, :meth:`bytes.count` and :meth:`bytearray.count`
methods for counting single characters.
```

I think we can keep it simple. You can add a WhatsNew entry with more information and indicate that this only for the case when *maxcount* is not specified.

Author: Maxcount is not a parameter for count. It is only used by internal function calls. So your suggestion works perfectly.

Reviewer 2: Oh yes you're right. All's good then!
